pywbemtools/pywbemcli/_pywbemcli_operations.py

```diff
@@ -288,41 +290,240 @@ def PyWbemcliQueryInstances(self, FilterQueryLanguage, FilterQuery,
+                    md5.update(file_source)
+            # Add the cache ID, so that manual tweaks on the cache files
+            # invalidates the cache.
+            md5.update(_ensure_bytes(cache_id))
+            new_md5_value = md5.hexdigest()
+
+            # Flag indicating that the mock environment needs to be built
+            # (or re-built). If False, the mock environment cache can be used.
+            need_rebuild = False
+
+            # Determine whether the mock environment needs to be rebuilt based
+            # on the (non-)existence of the cache files.
+            if not os.path.isfile(pickle_file) or not os.path.isfile(md5_file):
+                if verbose:
+                    click.echo("Mock environment for connection definition "
+                               "'{}' will be built because it was not cached.".
+                               format(connection_name))
+                need_rebuild = True
+
+            # Determine whether the mock environment needs to be rebuilt based
+            # on the MD5 hash value of the input file content.
+            if not need_rebuild:
+                with open(md5_file, 'r') as fp:
+                    cached_md5_value = fp.read()
+                if new_md5_value != cached_md5_value:
+                    if verbose:
+                        click.echo("Mock environment for connection "
+                                   "definition '{}' is cached but will be "
+                                   "rebuilt because the mock files have "
+                                   "changed.".format(connection_name))
+                    need_rebuild = True
+
+        else:
+            # No connections file context.
+
+            if verbose:
+                click.echo("Mock environment for connection definition '{}' "
+                           "will be built because no connections file is "
+                           "known.".format(connection_name))
+            need_rebuild = True
+
+        if need_rebuild:
+            try:
+                self._build_mockenv(server, file_path_list, verbose)
+            except mockscripts.NotCacheable as exc:
+                if verbose:
+                    click.echo("Mock environment for connection definition "
+                               "'{}' will be built because it is not "
+                               "cacheable: {}.".format(connection_name, exc))
+            else:
+                if connections_file:
+                    self._dump_mockenv(pickle_file)
+                    with open(md5_file, 'w') as fp:
+                        fp.write(new_md5_value)
+                    if verbose:
+                        click.echo("Mock environment for connection "
+                                   "definition '{}' has been written to "
+                                   "cache.".format(connection_name))
+        else:
+            # If we can use the cache, there was always a connections file
```

Comment:
I assume this comment is meant to say "There must be a connections file for caching to occur"

Reply:
Close but not exactly. If you want to phrase it this way around, then it would say: "There must be a connections file when no rebuild was needed."

DONE - Changed it to: "When no rebuild was needed, there must have been a connections file set."